### PR TITLE
python311Packages.devito: 4.8.8 -> 4.8.9

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Changelog: https://github.com/devitocodes/devito/releases/tag/v4.8.9

cc @AtilaSaraiva 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 326125` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.devito</li>
    <li>python311Packages.devito.dist</li>
    <li>python312Packages.devito</li>
    <li>python312Packages.devito.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 326125` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.devito</li>
    <li>python311Packages.devito.dist</li>
    <li>python312Packages.devito</li>
    <li>python312Packages.devito.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 326125` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.devito</li>
    <li>python311Packages.devito.dist</li>
    <li>python312Packages.devito</li>
    <li>python312Packages.devito.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 326125` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.devito</li>
    <li>python311Packages.devito.dist</li>
    <li>python312Packages.devito</li>
    <li>python312Packages.devito.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 326125` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.devito</li>
    <li>python311Packages.devito.dist</li>
    <li>python312Packages.devito</li>
    <li>python312Packages.devito.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/devito/default.nix [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
 buildPythonPackage rec {
   pname = "devito";
-  version = "4.8.8";
+  version = "4.8.9";
   format = "setuptools";
```
Comment (natsukium) [inline pkgs/development/python-modules/devito/default.nix:29]: Could you add `pyproject = true` and `build-system = [ setuptools ]`?

**Review by natsukium — APPROVED** [review]
The diff itself looks fine.

File: pkgs/development/python-modules/devito/default.nix (hunk 1 above)
Comment (GaetanLepage) [reply pkgs/development/python-modules/devito/default.nix:29]: I have already tried it and it doesn't work unfortunately
```
Checking runtime dependencies for devito-4.8.9-py3-none-any.whl
  + Exception Group Traceback (most recent call last):
  |   File "/nix/store/gdxd1rn8zgjjw08ijghvpazj18n4sjl6-python-runtime-deps-check-hook.py", line 93, in <module>
  |     metadata = get_metadata(args.wheel)
  |                ^^^^^^^^^^^^^^^^^^^^^^^^
  |   File "/nix/store/gdxd1rn8zgjjw08ijghvpazj18n4sjl6-python-runtime-deps-check-hook.py", line 58, in get_metadata
  |     metadata = Metadata.from_raw(raw)
  |                ^^^^^^^^^^^^^^^^^^^^^^
  |   File "/nix/store/mjb4mvqf4rmfam62w3j4094ghadgwvxp-python3.12-packaging-24.1/lib/python3.12/site-packages/packaging/metadata.py", line 701, in from_raw
  |     raise ExceptionGroup("invalid metadata", exceptions)
  | ExceptionGroup: invalid metadata (1 sub-exception)
  +-+---------------- 1 ----------------
    | Traceback (most recent call last):
    |   File "/nix/store/mjb4mvqf4rmfam62w3j4094ghadgwvxp-python3.12-packaging-24.1/lib/python3.12/site-packages/packaging/metadata.py", line 607, in _process_provides_extra
    |     normalized_names.append(utils.canonicalize_name(name, validate=True))
    |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |   File "/nix/store/mjb4mvqf4rmfam62w3j4094ghadgwvxp-python3.12-packaging-24.1/lib/python3.12/site-packages/packaging/utils.py", line 47, in canonicalize_name
    |     raise InvalidName(f"name is invalid: {name!r}")
    | packaging.utils.InvalidName: name is invalid: 'python_version_3.8_'
    | 
    | The above exception was the direct cause of the following exception:
    | 
    | Traceback (most recent call last):
    |   File "/nix/store/mjb4mvqf4rmfam62w3j4094ghadgwvxp-python3.12-packaging-24.1/lib/python3.12/site-packages/packaging/metadata.py", line 696, in from_raw
    |     getattr(ins, key)
    |   File "/nix/store/mjb4mvqf4rmfam62w3j4094ghadgwvxp-python3.12-packaging-24.1/lib/python3.12/site-packages/packaging/metadata.py", line 504, in __get__
    |     value = converter(value)
    |             ^^^^^^^^^^^^^^^^
    |   File "/nix/store/mjb4mvqf4rmfam62w3j4094ghadgwvxp-python3.12-packaging-24.1/lib/python3.12/site-packages/packaging/metadata.py", line 609, in _process_provides_extra
    |     raise self._invalid_metadata(
    | packaging.metadata.InvalidMetadata: 'python_version_3.8_' is invalid for 'provides-extra'
    +------------------------------------
```

Comment (natsukium) [reply pkgs/development/python-modules/devito/default.nix:29]: Oh, sorry. I think we had the same conversation in the previous update.

**Pull request merged**